lib/sites/flipkart.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const siteUtils = require('../site-utils');
+const logger = require('../logger')();
+
+class FlipkartSite {
+  constructor(uri, config) {
+    if (!FlipkartSite.isSite(uri)) {
+      throw new Error('invalid uri for Flipkart: %s', uri);
+    }
+
+    this._uri = uri;
+
+    // attempt to find the API key
+    if (!config.headers) {
+      const errorMessage = 'Flipkart cannot be called without an valid headers';
+      logger.error(errorMessage);
+      throw new Error(errorMessage);
+    }
+  }
+
+  getURIForPageData() {
+    const pid = /pid=([^&]*)/.exec(this._uri);
+    if (!pid || !pid[1]) {
+      logger.error('Could not detect the PID from the URL');
+      return null;
+    }
+    /* eslint-disable prefer-template */
+    return 'https://affiliate-api.flipkart.net/affiliate/product/json?' +
+     'id=' + pid[1];
+    /* eslint-enable prefer-template */
```

Comment:
I'm curious as to why the linter was disabled here, were you unable to use the template string syntax?